src/Shared/LogMessagePacketBase.cs

```diff
@@ -667,171 +711,7 @@ private BuildEventArgs GetBuildEventArgFromId()
-            {
-                return LoggingEventType.UninitializedPropertyRead;
-            }
-            else if (eventType == typeof(GeneratedFileUsedEventArgs))
-            {
-                return LoggingEventType.GeneratedFileUsedEvent;
-            }
-            else if (eventType == typeof(BuildCheckResultMessage))
-            {
-                return LoggingEventType.BuildCheckMessageEvent;
-            }
-            else if (eventType == typeof(BuildCheckResultWarning))
-            {
-                return LoggingEventType.BuildCheckWarningEvent;
-            }
-            else if (eventType == typeof(BuildCheckResultError))
-            {
-                return LoggingEventType.BuildCheckErrorEvent;
-            }
-            else if (eventType == typeof(BuildCheckAcquisitionEventArgs))
-            {
-                return LoggingEventType.BuildCheckAcquisitionEvent;
-            }
-            else if (eventType == typeof(BuildCheckTracingEventArgs))
-            {
-                return LoggingEventType.BuildCheckTracingEvent;
-            }
-#endif
-            else if (eventType == typeof(TargetStartedEventArgs))
-            {
-                return LoggingEventType.TargetStartedEvent;
-            }
-            else if (eventType == typeof(TargetFinishedEventArgs))
-            {
-                return LoggingEventType.TargetFinishedEvent;
-            }
-            else if (eventType == typeof(TaskStartedEventArgs))
-            {
-                return LoggingEventType.TaskStartedEvent;
-            }
-            else if (eventType == typeof(TaskFinishedEventArgs))
-            {
-                return LoggingEventType.TaskFinishedEvent;
-            }
-            else if (eventType == typeof(BuildFinishedEventArgs))
-            {
-                return LoggingEventType.BuildFinishedEvent;
-            }
-            else if (eventType == typeof(BuildStartedEventArgs))
-            {
-                return LoggingEventType.BuildStartedEvent;
-            }
-            else if (eventType == typeof(BuildWarningEventArgs))
-            {
-                return LoggingEventType.BuildWarningEvent;
-            }
-            else if (eventType == typeof(BuildErrorEventArgs))
-            {
-                return LoggingEventType.BuildErrorEvent;
-            }
-            else if (eventType == typeof(EnvironmentVariableReadEventArgs))
-            {
-                return LoggingEventType.EnvironmentVariableReadEvent;
-            }
-            else if (eventType == typeof(ResponseFileUsedEventArgs))
-            {
-                return LoggingEventType.ResponseFileUsedEvent;
-            }
-            else
-            {
-                return LoggingEventType.CustomEvent;
-            }
+            return EventTypeToLoggingEventTypeMap.TryGetValue(eventType, out var loggingEventType) ? loggingEventType : LoggingEventType.CustomEvent;
```

Comment:
Adding a big dictionary here is an extra allocation. I'd rather just see a switch statement like:
```C#
return eventType switch {
    BuildMessageEventArgs => LoggingEventType.BuildMessageEvent,
    TaskCommandLineEventArgs => LoggingEventType.TaskCommandLineEvent,
    ...
    _ => LoggingEventType.CustomEvent
};

Reply:
A single static dictionary (ideally `FrozenDictionary` on net9) allocation isn't something I'd worry about. The switch is nice and clear but does appear (at least at the moment: https://github.com/dotnet/runtime/issues/12260) to compile down to a [linear list of checks](https://sharplab.io/#v2:EYLgHgbALAPgAgJgIwFgBQcDMACR2DC2A3utmbjnEgAzYCyAFAEICGAzgKbbACUxp5QXADs3bGwDuASwAuAYwAW/NINXkAIkmwBeAHzYARJoMAaAWsHqEO/UYSnzFsupx7DLhyqcaoN91E9vDQBWPyNgwKDsdQgwmMig9VE3I2EE73UADjjM9Kd1AE44gryLAH0wgH5q0rIAXwBuRzr0FrR0LFxrVk5iNo7Ka01sEGweriI2zrwrEbH2CanB6JxR8b6BrujfNYWNjGX1UN3eyc2Z2JPF86HRK/3poez7s4Otwrn1yaA=).

Using a dictionary here seems like it'd be reasonable for > 10 types: https://github.com/dotnet/runtime/issues/12260#issuecomment-473691570

Reply:
It looks like jkotas suggested a hashtable. Not a big difference between that and a dictionary for this application, though, as far as I know.

Reply:
I guess the other half of this is asking what the real goal of this change is, since I think switch statements are always beautiful, and I don't imagine this is a perf bottleneck

Reply:
The primary objective was to simplify the lengthy and difficult-to-read if/else statements in the method. Regarding the use of switch statements or dictionaries, could you please share your thoughts on the current implementation? If you have any concerns, I would appreciate hearing your reasoning.

Reply:
It sounds like rainersigwald likes it better, so I don't have any real concerns. I still prefer a switch, but either is fine.